backend/src/main/java/com/github/nickklock/backend/controllers/UserController.java

```diff
@@ -19,12 +18,17 @@ public UserController(UserService userService) {
     }
 
     @PostMapping
-    public ResponseEntity<UserSpot> add(@RequestBody UserAuth userAuth) {
-        return new ResponseEntity<>(userService.createNewUser(userAuth), HttpStatus.CREATED);
+    public ResponseEntity<UserSpot> add(@RequestBody NewUserRequest newUserRequest) {
+        return new ResponseEntity<>(userService.createNewUser(newUserRequest), HttpStatus.CREATED);
     }
 
     @GetMapping("/me")
-    public String me() {
-        return SecurityContextHolder.getContext().getAuthentication().getName();
+    public ResponseEntity<UserSpot> me() {
+        return new ResponseEntity<>(userService.getUserSpotBySecurityContext(), HttpStatus.OK);
+    }
```

Comment:
You can use the factory-pattern style `.of()` for response entity. If the optional is present, it will be 200 OK, if not 404 NOT FOUND. Pretty handy for get mappings.

```suggestion
    public ResponseEntity<UserSpot> me() {
        return ResponseEntity.of(userService.getUserSpotBySecurityContext());
    }
```